File: benchmarks/poly/jsondeserialize/jsondeserialize.php
<?php

function main() {
    $_t0 = hrtime(true);
    $n = 400000;
    $checksum = 0;
    $total_name_len = 0;

    for ($i = 0; $i < $n; $i++) {
        $line = '{"id":' . $i . ',"name":"user' . $i . '","score":' . ($i % 100) . '}';

        $id_start = strpos($line, ":") + 1;
        $id_end = strpos($line, ",");
        $id_val = (int)substr($line, $id_start, $id_end - $id_start);

        $name_prefix = '"name":"';
        $name_start = strpos($line, $name_prefix) + strlen($name_prefix);
        $name_end = strpos($line, '"', $name_start);
        $name_val = substr($line, $name_start, $name_end - $name_start);

        $score_start = strrpos($line, ":") + 1;
        $score_end = strrpos($line, "}");
        $score_val = (int)substr($line, $score_start, $score_end - $score_start);

        $checksum = ($checksum * 31 + $id_val + $score_val) % 1000000007;
        $total_name_len += strlen($name_val);
    }

    echo $checksum . "\n";
    echo $total_name_len . "\n";
    fwrite(STDERR, "bench_ns=" . (hrtime(true) - $_t0) . "\n");
}

main();

File: benchmarks/poly/jsondeserialize/jsondeserialize.lua
local function main()
    local _t0 = os.clock()
    local n = 400000
    local checksum = 0
    local total_name_len = 0

    for i = 0, n - 1 do
        local line = '{"id":' .. i .. ',"name":"user' .. i .. '","score":' .. (i % 100) .. "}"

        local id_str, name_val, score_str = line:match('"id":(%d+),"name":"(user%d+)","score":(%d+)}')
        local id_val = tonumber(id_str)
        local score_val = tonumber(score_str)

        checksum = (checksum * 31 + id_val + score_val) % 1000000007
        total_name_len = total_name_len + #name_val
    end

    print(checksum)
    print(total_name_len)
    io.stderr:write(string.format("bench_ns=%.0f\n", (os.clock() - _t0) * 1e9))
end

main()
